src/porepy/grids/partition.py

```diff
@@ -484,7 +484,13 @@ def __extract_submatrix(mat, ind):
     cols = sub_mat.indptr
     data = sub_mat.data
     unique_rows, rows_sub = np.unique(sub_mat.indices, return_inverse=True)
-    return sps.csc_matrix((data, rows_sub, cols)), unique_rows
+    shape = (unique_rows.size, cols.size - 1)
+    if shape[0]!=0:
+        A = sps.csc_matrix((data, rows_sub, cols))
+        if np.any(A.shape != shape):
+            import pdb; pdb.set_trace()
```

Comment:
Something like this, although the question marks should be replaced by an informative message. Also please add a comment on whath the outer if does.
```suggestion
            raise ValueError("???")
```

Reply:
Both the inner and outer loop was for debugging only. I removed both